### PR TITLE
Add spinner while waiting for location and ranking retrieval

### PR DESCRIPTION
**Pull request by helenatran**
Add a spinner while waiting on the fetching of the location and the ranking.

Note: I haven't added the spinner for the recommendation because how the hidden gems are being retrieved will soon change; so I didn't want to have a spinner working for the current version but not the newer version of the code. After the code change, we should be able to implement the spinner in the same was as for the ranking (while waiting on the fetch).

Screen Shot 2021-02-12 at 10 53 46 am

Screen Shot 2021-02-12 at 10 54 15 am (2)

### CODE REVIEW

**Review by jamila-sabaz — APPROVED** [review]

File: src/main/webapp/src/app/location/location.component.ts [hunk 1]
```diff
@@ -50,6 +51,7 @@ export class LocationComponent{
   }
 
   enableNavigatorLocation(): void {
+    this.isLocationButtonClicked = true;
```
Comment (lorikeet-s) [inline src/main/webapp/src/app/location/location.component.ts:54]: Does anything ever set this back to false? or is the component just unloaded?
Could you add a comment explaining this a bit?

**Review by lorikeet-s — CHANGES_REQUESTED** [review]
Could you add a screenshot to the PR?

File: src/main/webapp/src/app/location/location.component.ts (hunk 1 above)
Comment (helenatran) [reply src/main/webapp/src/app/location/location.component.ts:56]: When the user clicks on the button to retrieve their location and the location has been fetched, it redirects the user to the '/home' page; so the spinner goes away. Is it ok to keep it like that?

**Comment by helenatran** [comment]
Update: I have added a spinner for map as well. I tried writing the tests to check the spinners but got some errors. I would like to move on and be able to update other things before the presentation, so added some todo comments to write the tests.
On another note, "google is undefined" errors are completely gone, but we still sometimes get the error "Error: Uncaught (in promise): Error: Cannot match any routes. URL Segment: 'home'" from Location component from time to time (weirdly, it doesn't always appear!)

**Comment by helenatran** [comment]
@lorikeet-s screenshots have been added to the PR

**Review by lorikeet-s — APPROVED** [review]

**Pull request merged**